### ISSUE TITLE
RCS1036 False Positive

### ISSUE THREAD

**Issue by nnpcYvIVl** [JosefPihrt/Roslynator#777]
Roslynator version 3.1.0 throws RCS1036 on `M1()` in the following code. Is this a false positive?

```
namespace RCS1036_False_Positive
{
    public static class Class
    {
        private static int M1() => default; /// <see cref="M2()"/>

        private static int M2() => default;
    }
}
```

**Comment by JosefPihrt** [comment]
Is this a real code sample? I'm asking because the documentation comment belongs to the second declaration so the code should look like this:

```cs
namespace RCS1036_False_Positive
{
    public static class Class
    {
        private static int M1() => default;

        /// <see cref="M2()"/>

        private static int M2() => default;
    }
}
```
and in this case it is reasonable to remove empty line between documentation comment and declaration.

**Comment by nnpcYvIVl** [comment]
@JosefPihrt It is a simplification of a real code sample. The comment belongs to M1 and refers to M2. M2 is far away from M1 so the analyzer is prompting to remove the blank line between M1 and another unrelated method.

Here is a better approximation with M1 referring to M9.
```
namespace RCS1036_False_Positive
{
    public static class Class
    {
        private static int M1() => default; /// <see cref="M9()"/>
        
        private static int M2() => default;
        
        private static int M3() => default;

        private static int M9() => default;
    }
}
```

**Comment by JosefPihrt** [comment]
Documentation comment must be placed before a declaration. So Roslyn sees your documentation comment as a documentation comment for `M2` not `M1`. This is standard behavior.

![image](https://user-images.githubusercontent.com/11931318/109729873-d7397d00-7bb8-11eb-95e6-8a45832b9c32.png)

If your comment belongs to the `M1` it should be placed before a declaration.

```cs
namespace RCS1036_False_Positive
{
    public static class Class
    {
        /// <see cref="M9()"/>
        private static int M1() => default;
        
        private static int M2() => default;
        
        private static int M3() => default;

        private static int M9() => default;
    }
}
```

**Comment by nnpcYvIVl** [comment]
Thanks for clearing that up!
